packages/react-components/react-dialog/src/utils/useFocusFirstElement.ts

```diff
@@ -21,10 +21,12 @@ export function useFocusFirstElement(open: boolean, modalType: DialogModalType)
     const element = dialogRef.current && findFirstFocusable(dialogRef.current);
     if (element) {
       element.focus();
-    } else if (process.env.NODE_ENV !== 'production') {
-      triggerRef.current?.blur();
-      // eslint-disable-next-line no-console
-      console.warn('A Dialog should have at least one focusable element inside DialogSurface');
+    } else {
+      dialogRef.current?.focus(); // https://github.com/microsoft/fluentui/issues/25150
+      if (process.env.NODE_ENV !== 'production') {
+        // eslint-disable-next-line no-console
+        console.warn('A Dialog should have at least one focusable element inside DialogSurface');
```

Comment:
nit: Let's add a pointer for consumers and call to action on _how to solve_ the issue? Something like that? ⬇️

```suggestion
        console.warn('@fluentui/react-dialog: a Dialog should have at least one focusable element inside DialogSurface. Please add ____TODO___');
```